agnpy/spectra/tests/test_spectra.py

```diff
@@ -817,4 +824,107 @@ def test_compare_calculation_with_calculation_split_into_two(self):
+            mass=m_e,
+        )
+
+        blob = Blob(r_b, n_e=n_e)
+        synch = Synchrotron(blob)
+        ssc = SynchrotronSelfCompton(blob)
+
+        time = 60 * u.s
+        intervals = 600
+
+        electrons_before = n_e.integrate()
+
+        blob.n_e_time_evolution(synch.electron_energy_loss_rate, time, subintervals_count=intervals)
+        electrons_after_sync = blob.n_e.integrate()
+
+        blob.n_e = n_e
+        blob.n_e_time_evolution(ssc.electron_energy_loss_rate_thomson, time, subintervals_count=intervals)
+        electrons_after_ssc_th = blob.n_e.integrate()
+
+        blob.n_e = n_e
+        blob.n_e_time_evolution(ssc.electron_energy_loss_rate, time, subintervals_count=intervals)
+        electrons_after_ssc = blob.n_e.integrate()
+
+        assert u.isclose(electrons_before, electrons_after_sync, rtol=0.005)
+        assert u.isclose(electrons_before, electrons_after_ssc_th, rtol=0.005)
+        assert u.isclose(electrons_before, electrons_after_ssc, rtol=0.005)
+
+    def test_ssc_losses_in_thompson_limit(self):
+        r_b = 1e16 * u.cm
+        n_e = PowerLaw.from_total_energy(
+            1e30 * u.erg,
+            4 / 3 * np.pi * r_b ** 3,
+            p=2.8,
+            gamma_min=1e1,
+            gamma_max=1e3,
+            mass=m_e,
+        )
+
+        blob = Blob(r_b, n_e=n_e)
+        ssc = SynchrotronSelfCompton(blob)
 
+        thomson = ssc.electron_energy_loss_rate_thomson(np.logspace(1, 3, 200)).to("erg/s")
+        ssc = ssc.electron_energy_loss_rate(np.logspace(1, 3, 200)).to("erg/s")
+
+        assert u.allclose(thomson, ssc, 0.01)
+
+    @pytest.mark.parametrize("p", [0.5, 1.2, 2.4])
+    @pytest.mark.parametrize("time_and_steps", [(1, 10), (60, 60), (200, 100)])
+    def test_compare_numerical_results_with_analytical_calculation_ssc_thomson(self, p, time_and_steps):
+        """Test time evolution of spectral electron density for SSC energy losses in Thomson range.
+         Similar to the corresponding test for Synchrotron losses."""
+        time = time_and_steps[0] * u.s
+        steps = time_and_steps[1]
+        r_b = 1e16 * u.cm
+        gamma_min = 1e1
+        gamma_max = 1e4
+        n_e = PowerLaw.from_total_energy(
+            1e20 * u.erg,
+            4 / 3 * np.pi * r_b ** 3,
+            p=p,
+            gamma_min=gamma_min,
+            gamma_max=gamma_max,
+            mass=m_e,
+        )
+        k = n_e.k
+        blob = Blob(r_b, n_e=n_e)
+        initial_gammas = blob.gamma_e
+
+        ssc = SynchrotronSelfCompton(blob)
+        blob.n_e_time_evolution(ssc.electron_energy_loss_rate, time, subintervals_count=steps)
+        evaluated_n_e: InterpolatedDistribution = blob.n_e
+
+        def gamma_before(gamma_after_time, before_time):
+            coef = ssc._electron_energy_loss_thomson_formula_prefactor() / (m_e * c ** 2)
+            return 1 / ((1 / gamma_after_time) - before_time * coef)
+
+        def integral_analytical(gamma_bottom, gamma_up):
```

Comment:
I think this function is being used in two different test functions - it can be extracted out of the functions to be included only once (and I would change the name to intergral_powerlaw or something like this

Reply:
It turned out there is already sch function in this file, so i removed mine.